services/motion/builtin/move_request.go

```diff
@@ -207,12 +215,28 @@ func (mr *moveRequest) obstaclesIntersectPlan(
 				return state.ExecuteResponse{}, err
 			}
 
+			// determine transform of camera to base
```

Comment:
Is this fixing a bug in `MoveOnGlobe`'s obstacle avoidance? If so, what is the bug & have you run a hardware test to confirm it now works as expected?

Reply:
> Is this fixing a bug in MoveOnGlobe's obstacle avoidance? If so, what is the bug 

Yes this is a bug.
Below I describe a scenario where this occurs.
Let a camera return detections in the xz plane.
Suppose the camera returns a detection at position `(a,0,b)`
let our localizer tell us we are at some `(x,y,z)` position

originally we did
`compose((a,0,b),(x,y,z))` yielding `(a+x, y, b +z)`

instead what we should do first is convert `(a,0,b)` into the base coordinate system
giving us `(a',b',0)` we then compose with our current location
`compose((a',b',0),(x,y,z))` yielding `(a'+x, b'+y, z)`

> have you run a hardware test to confirm it now works as expected?

I did not run a hardware test on this. I will do so when I am back in the office.